### ISSUE TITLE
What about killing /C

### ISSUE THREAD

**Issue by Maijin** [radareorg/radare2#8989]
What about killing /C

- It is not tested
- It does not work
- Yara provides better crypto detections
- Radare2 have several Yara plugins

**Comment by radare** [comment]
Yara Cant implement /C for aes expanded keys. Its not about constants. And if we kill that we should provide proper multikey search support like it was in r1. As well as providing sdb or plaintext files with constant tables and such for different algorithms

**Comment by Maijin** [comment]
well then provide with a working example please as currently there is none and it's probably completly dead :/

**Comment by radare** [comment]
i dont think its dead t all, the algo is quite straight forward and the search api didnt changed in several years

the expanded aes key search thing was done for the wii jb

**Comment by xarkes** [comment]
Fix this please.

**Comment by radare** [comment]
Fix what

**Comment by XVilka** [comment]
Do we have a testcase for this? And close this bug after.

**Comment by Maijin** [comment]
@XVilka no because it doesn't work

**Comment by radare** [comment]
what doesnt works?

**Comment by radare** [comment]
which post.

**Comment by Maijin** [comment]
The issue title " What about killing /C #8989 "

**Comment by Maijin** [comment]
Using AESCrypt2 which contains S-box:

git clone https://github.com/palmerc/AESCrypt2 && cd AESCrypt2 && make

r2 aescrypt2 or r2 aescrypt2.exe
(tried various e search.in, e search.from etc.) /Ca
```
[0x100001090]> /Ca
Searching 0 byte in [0x100000f60-0x100009ce6]
hits: 0
Searching 0 byte in [0x100009cf0-0x100009f92]
hits: 0
Searching 0 byte in [0x100009f94-0x100009ffc]
hits: 0
Searching 0 byte in [0x10000a000-0x10000a0c8]
hits: 0
Searching 0 byte in [0x10000a0d0-0x100011170]
hits: 0
```

Compiled file can be found here - [aescrypt2.zip](https://github.com/radare/radare2/files/2282823/aescrypt2.zip)

**Comment by radare** [comment]
U r doing it wrong

/Ca searchs for EXPANDED AES KEYS in memory

Not crypto tables of anything in static.

Put a breakpoint before decrypting a blob and dump the ram. You may get a hit in there.

**Comment by Maijin** [comment]
This neat tool is in the public domain and works nicely.
Could probably be integrate to replace /Ca
https://github.com/mmozeiko/aes-finder

For /Cr this MIT licensed tool is worth integrating:
https://github.com/cryptolok/CryKeX
I haven't tested it though.

from @0ki

**Comment by brainstorm** [comment]
I agree with @Maijin and the thread in general, `/Ca` is very much not intuitive (see https://github.com/radare/radare2/pull/12580 and https://github.com/radare/radare2book/pull/168).

I don't know about the other `/C`'s but I would just write a section on the radarebook on how to search all crypto stuff with external tools yet using radare to collect the hits and label locations/xrefs, IMHO.

**Comment by Maijin** [comment]
@brainstorm 

We can maybe a section on yara with r2 using the crypto signatures from yara-rules.

**Comment by Maijin** [comment]
I know. That was for the standard static AES crypto search which most people are looking for.

**Comment by brainstorm** [comment]
Ok ppl, I tried taking a look at Yara but I found this dog breakfast (`yara-r2` vs `r2yara` vs `yara` vs **not** `yara3`):

```bash
$ r2pm -s yara
r2yara     	 [syspkg] r2 module for yara. Powerful of r2 functionalities in Yara	
yara     	 [syspkg] yara library and commandline tools from git			
yara-r2     	 [r2-core] yara plugin for radare2
```

And since docs refer to `yara3`...:

https://github.com/radare/radare2/blob/master/doc/yara.md

... what's the current state of the art w.r.t radare2 and yara? How do people use it nowadays before starting to write already deprecated stuff on the radare2book?

Cheers!
